server/container_create_linux.go

```diff
@@ -1249,15 +1264,63 @@ func (s *Server) mountImage(ctx context.Context, imageRef, mountLabel string) (h
 	log.Debugf(ctx, "Image ID to mount: %v", id)
 
 	options := []string{"ro", "noexec", "nosuid", "nodev"}
-	mountPoint, err := s.Store().MountImage(id, options, mountLabel)
+	mountPoint, err := s.Store().MountImage(id, options, "")
 	if err != nil {
 		return "", "", fmt.Errorf("mount storage: %w", err)
 	}
-
 	log.Infof(ctx, "Image mounted to: %s", mountPoint)
+
+	const overlay = "overlay"
+	specgen.AddMount(rspec.Mount{
+		Type:        overlay,
+		Source:      overlay,
+		Destination: containerPath,
+		Options: []string{
+			"lowerdir=" + mountPoint + ":" + imageVolumesPath,
+		},
+	})
```

Comment:
@haircommander is that enough to propagate the right label to the container when we call afterwards:

https://github.com/cri-o/cri-o/blob/c940a11495c28f0f1cfeeeb461c805796944c2af/server/container_create_linux.go#L812-L813

Reply:
At least it works in my local VM-based test.